rig/machine_control/machine_controller.py

```diff
@@ -917,6 +935,30 @@ def send_signal(self, signal, app_id=Required):
         self._send_scp(0, 0, 0, SCPCommands.signal, arg1, arg2, arg3)
 
     @ContextMixin.use_contextual_arguments
+    def app_start(self, app_id=Required):
+        """Transmit a signal to start the application."""
+        self.send_signal(consts.AppSignal.start, app_id)
+
+    @ContextMixin.use_contextual_arguments
+    def app_stop(self, app_id=Required):
+        """Transmit a signal to stop the application."""
+        self.send_signal(consts.AppSignal.stop, app_id)
+
+    @ContextMixin.use_contextual_arguments
+    def sync0(self, app_id=Required):
```

Comment:
Actually, I totally get the use for that - d'oh. There's some really cool stuff we can do here because we can grab the full core state (like write the reason for any RTEs out).  I'll get on this tomorrow.